### PR TITLE
feat: provide OpenAI compatible API for chat

### PR DESCRIPTION
**Pull request by kemingy**

### CODE REVIEW

**Comment by kemingy** [comment]
I'll test this on my CPU machine.

File: main.py [hunk 1]
```diff
@@ -1,36 +1,92 @@
...
+    CompletionResponse,
+    TokenUsage,
+    ChatChoice,
+    ChatMessage,
+    Role,
+)
 
 logger = get_logger()
+TOKENIZER = os.environ.get("TOKENIZER", "THUDM/chatglm-6b-int4")
+MODEL = os.environ.get("MODEL", "THUDM/chatglm-6b-int4")
```
Comment (gaocegege) [inline main.py:19]: Could we add a prefix MODELZ?

Comment (kemingy) [reply main.py:19]: Sure

**Comment by kemingy** [comment]
Test:

```console
❯ http :8000/inference model=test messages:='[{"content": "who are you"}]'
HTTP/1.1 200 OK
content-length: 358
date: Fri, 19 May 2023 04:25:15 GMT
server: mosec/0.6.6

{"id":"THUDM/chatglm-6b-int4","object":"chat","created":"2023-05-19T04:25:15.229614","choices":[{"message":{"content":"I am an AI assistant named ChatGLM-6B. I was developed based on the language model jointly trained by T","role":"assistant","name":""},"index":0,"finish_reason":"stop"}],"usage":{"prompt_tokens":5,"completion_tokens":25,"total_tokens":30}}
```

**Comment by kemingy** [comment]
It's ready to review. @gaocegege PTAL

**Comment by gaocegege** [comment]
- [ ] Content-Type application/json

**Comment by kemingy** [comment]
> * [ ]  Content-Type application/json

- refer to https://github.com/mosecorg/mosec/issues/357

**Pull request merged**